### PR TITLE
add typings for browser-image-compression

### PR DESCRIPTION
**Pull request by Donaldcwl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/browser-image-compression/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    exifOrientation?: number;
+    /** A function takes one progress argument (progress from 0 to 100) */
+    onProgress?: (progress: number) => void;
+    /** Default to be the original mime type from the image file */
+    fileType?: string;
+}
+
+declare function imageCompression(image: File | Blob, options: Options): Promise<File | Blob>;
+
+export = imageCompression;
```
Comment (peterblazejewicz) [inline types/browser-image-compression/index.d.ts]:
```suggestion
declare function imageCompression(image: File | Blob, options: Options): Promise<File | Blob>;

export as namespace imageCompression;

export = imageCompression;
```
what about the browser?
I would like to use sample code stright from the readme;
```ts
async function handleImageUpload({ target }: { target: EventTarget & { files: FileList } }) {
    const imageFile = target.files[0];
    console.log('originalFile instanceof Blob', imageFile instanceof Blob); // true
    console.log(`originalFile size ${imageFile.size / 1024 / 1024} MB`);

    const options = {
        maxSizeMB: 1,
        maxWidthOrHeight: 1920,
        useWebWorker: true,
    };
    try {
        const compressedFile = await imageCompression(imageFile, options);
        console.log('compressedFile instanceof Blob', compressedFile instanceof Blob); // true
        console.log(`compressedFile size ${compressedFile.size / 1024 / 1024} MB`); // smaller than maxSizeMB

        // await uploadToServer(compressedFile);
    } catch (error) {
        console.log(error);
    }
}
```

Comment (Donaldcwl) [reply types/browser-image-compression/index.d.ts]: Thank you.

**Review by peterblazejewicz — APPROVED** [review]
LGTM
@Donaldcwl thx for contribution!
Thanks for adding JSDoc comments, useful for the package consumption by users!

**Comment by Donaldcwl** [comment]
May I ask when will this PR be accepted?

**Comment by Donaldcwl** [comment]
When will the PR be merged? Thanks!

**Comment by Donaldcwl** [comment]
> We've gotten sign-off from a reviewer 👏. A maintainer will soon review this PR and merge it if there are no issues. If it shouldn't be merged yet, please leave a comment saying so and we'll wait. Thank you for contributing to DefinitelyTyped!
When will the PR be merged? Thanks!

**Comment by peterblazejewicz** [comment]
@Donaldcwl I hope soon, I understand people are busy with a lot of things atm. Let's wait for a little while, pls!

**Pull request merged**